src/frontend/sharedComponents/ErrorModal.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import {BottomSheetModal, useBottomSheetModal} from './BottomSheetModal';
+import {Button} from './Button';
+import {StyleSheet, View} from 'react-native';
+import {LogoWithErrorIcon} from './LogoWithErrorIcon';
+import {Text} from './Text';
+import {defineMessages, useIntl} from 'react-intl';
+
+const m = defineMessages({
+  somethingWrong: {
+    id: 'sharedComponents.ErrorModal.somethingWrong',
+    defaultMessage: 'Something Went Wrong',
+  },
+  goBack: {
+    id: 'sharedComponents.ErrorModal.goBack',
+    defaultMessage: 'Go Back',
+  },
+  tryAgain: {
+    id: 'sharedComponents.ErrorModal.tryAgain',
+    defaultMessage: 'Please go back and try again',
+  },
+});
+
+type ErrorModalProps = Omit<
+  ReturnType<typeof useBottomSheetModal>,
+  'openSheet'
+> & {clearError?: () => void};
+
+/**
+ *
+ * should be used with 'useBottomSheetModal()' Hook
+ */
+export const ErrorModal = ({
+  clearError,
+  closeSheet,
```

Comment:
seems like it would make sense to specify a single prop instead of having two separate props in this case and defining the `handleGoBack()` function (which is the only thing that uses these props)

 e.g.

```tsx
const SomeComponent = () => {
  return <ErrorModal 
    onDismiss={() => {
      // Clear the error if necessary
      // Close the sheet
    }}
  />
}
```

Then this ErrorModal component only needs to pass that `onDimiss` prop directly to the `onPress` prop for the button.

Reply:
its need to take the `closeSheet` from the hook, and then the `clearError` needs to be defined by the api call. So we could pass an `onDismiss`, but we would have to make sure that the onDismiss includes the `closeSheet` function call. So it feels more intuitive to just have both props.

Reply:
fair enough!